indexer/src/test/java/org/uniprot/store/indexer/uniprotkb/InactiveEntryIndexJobIT.java

```diff
@@ -0,0 +1,123 @@
+import org.uniprot.store.search.SolrCollection;
+import org.uniprot.store.search.document.uniprot.UniProtDocument;
+
+/**
+ *
+ * @author jluo
+ * @date: 5 Sep 2019
+ *
+*/
+@ActiveProfiles(profiles = {"job", "offline"})
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = {SolrTestConfig.class, FakeIndexerSpringBootApplication.class, InactiveEntryIndexJob.class,
+                           InactiveEntryStep.class, ListenerConfig.class})
+@TestPropertySource(properties = {"uniprotkb.indexing.itemProcessorTaskExecutor.corePoolSize=1",
+        "uniprotkb.indexing.itemProcessorTaskExecutor.maxPoolSize=1",
+        "uniprotkb.indexing.itemWriterTaskExecutor.corePoolSize=1",
+        "uniprotkb.indexing.itemWriterTaskExecutor.maxPoolSize=1"})
+public class InactiveEntryIndexJobIT {
+	@Autowired
+    private JobLauncherTestUtils jobLauncher;
+    @Autowired
+    private UniProtSolrOperations solrOperations;
+    @Autowired
+    private UniProtKBIndexingProperties indexingProperties;
+
+    @Test
+    void testUniProtKBIndexingJob() throws Exception {
+        JobExecution jobExecution = jobLauncher.launchJob();
+        assertThat(jobExecution.getJobInstance().getJobName(), CoreMatchers.is(INACTIVEENTRY_INDEX_JOB));
+
+       Thread.sleep(5000);
+
+        BatchStatus status = jobExecution.getStatus();
+        assertThat(status, is(BatchStatus.COMPLETED));
+
+        Collection<StepExecution> stepExecutions = jobExecution.getStepExecutions();
+        assertThat(stepExecutions, hasSize(1));
+
+        checkInactiveEntryIndexingStep(jobExecution, stepExecutions);
+     
+    }
+
+    private void checkInactiveEntryIndexingStep(JobExecution jobExecution, Collection<StepExecution> stepExecutions)
+            throws IOException {
+        StepExecution kbIndexingStep = stepExecutions.stream()
+                .filter(step -> step.getStepName().equals(INACTIVEENTRY_INDEX_STEP))
+                .collect(Collectors.toList()).get(0);
+
+        assertThat(kbIndexingStep.getReadCount(), is(22));
+        checkWriteCount(jobExecution, CommonConstants.FAILED_ENTRIES_COUNT_KEY, 0);
+        checkWriteCount(jobExecution, CommonConstants.WRITTEN_ENTRIES_COUNT_KEY, 22);
+
+        // check that the accessions in the source file, are the ones that were written to Solr
+        Set<String> sourceAccessions = readSourceAccessions();
+        assertThat(sourceAccessions, hasSize(22));
+
+        Page<UniProtDocument> response = solrOperations
+                .query(SolrCollection.uniprot.name(), new SimpleQuery("*:*"), UniProtDocument.class);
+
+        assertThat(response, is(notNullValue()));
+        assertThat(response.getTotalElements(), is(22L));
+        Set<String> results = response.stream().map(doc -> doc.accession).collect(Collectors.toSet());
+        results.forEach(accession -> assertThat(sourceAccessions, hasItem(accession)) );
+       
+      
+//        assertThat(response.stream().map(doc -> doc.accession).collect(Collectors.toSet()),
```

Comment:
you can uncomment these lines if you replace lines 96-106:

```java
        SimpleQuery query = new SimpleQuery("*:*");
        query.setPageRequest(new PageRequest(0, 30));
        Page<UniProtDocument> response = solrOperations
                .query(SolrCollection.uniprot.name(), query, UniProtDocument.class);
        
        assertThat(response, is(notNullValue()));
        assertThat(response.getTotalElements(), is(22L));
        Set<String> results = response.stream().map(doc -> doc.accession).collect(Collectors.toSet());
        results.forEach(accession -> assertThat(sourceAccessions, hasItem(accession)) );
       
        assertThat(response.stream().map(doc -> doc.accession).collect(Collectors.toSet()),
                   is(sourceAccessions));
```

Reply:
good point

Reply:
PageRequest is a deprecated class